diff --git a/.gitattributes b/.gitattributes
index 23d8a9638..442980b7d 100644
--- a/.gitattributes
+++ b/.gitattributes
@@ -6,3 +6,5 @@ support/openharmony/**/*.ets linguist-language=ts
 # Denote all files that are truly binary and should not be modified.
 *.png binary
 *.jpg binary
+*.ico binary
+*.icns binary
diff --git a/resources/org.servo.Servo.desktop b/resources/org.roves.Roves.desktop
similarity index 91%
rename from resources/org.servo.Servo.desktop
rename to resources/org.roves.Roves.desktop
index 332b23dd8..04c652eff 100644
--- a/resources/org.servo.Servo.desktop
+++ b/resources/org.roves.Roves.desktop
@@ -1,10 +1,10 @@
-# You can use this file to get a menu entry on Linux and so you can pin Servo to your taskbar.
+# You can use this file to get a menu entry on Linux and so you can pin Roves to your taskbar.
 #
 # Steps:
 #   1. Create the applications folder if it doesn't exist:
 #           mkdir -p ~/.local/share/applications/
 #   2. Copy this file:
-#           cp org.servo.Servo.desktop ~/.local/share/applications
+#           cp org.roves.Roves.desktop ~/.local/share/applications
 #   3. Edit the copied file and replace the two SERVO_SRC_PATH occurrences
 #      with the path to the Servo sources.
 #   4. Create the icons folder if it doesn't exist:
@@ -13,7 +13,7 @@
 #          ln -s servo.svg ~/.local/share/icons/hicolor/scalable/apps/servo.svg
 [Desktop Entry]
 Version=1.0
-Name=Servo
+Name=Roves
 GenericName=Web Browser
 Comment=Browse the Web
 # TODO: Replace SERVO_SRC_PATH with the path to the Servo sources
