diff --git a/components/script/dom/indexeddb/idbrequest.rs b/components/script/dom/indexeddb/idbrequest.rs
index 0f00226aa3c..ea7bdf2d96e 100644
--- a/components/script/dom/indexeddb/idbrequest.rs
+++ b/components/script/dom/indexeddb/idbrequest.rs
@@ -8,10 +8,10 @@ use dom_struct::dom_struct;
 use js::context::JSContext;
 use js::conversions::ToJSValConvertible;
 use js::jsapi::Heap;
-use js::jsval::{DoubleValue, JSVal, ObjectValue, UndefinedValue};
+use js::jsval::{DoubleValue, JSVal, UndefinedValue};
 use js::rust::HandleValue;
 use profile_traits::generic_callback::GenericCallback;
-use script_bindings::reflector::{DomObject, dom_object_value, reflect_dom_object_with_cx};
+use script_bindings::reflector::{dom_object_value, reflect_dom_object_with_cx};
 use serde::{Deserialize, Serialize};
 use servo_base::generic_channel::GenericSend;
 use storage_traits::indexeddb::{
@@ -244,9 +244,9 @@ impl RequestListener {
                     if let Some(cursor) = cursor {
                         match cursor.downcast::<IDBCursorWithValue>() {
                             Some(cursor_with_value) => {
-                                answer.handle_mut().set(ObjectValue(
-                                    *cursor_with_value.reflector().get_jsobject(),
-                                ));
+                                answer
+                                    .handle_mut()
+                                    .set(dom_object_value(&*cursor_with_value));
                             },
                             None => {
                                 answer
diff --git a/components/script/dom/worklet/paintworkletglobalscope.rs b/components/script/dom/worklet/paintworkletglobalscope.rs
index 0c0c1297a71..215817bf69b 100644
--- a/components/script/dom/worklet/paintworkletglobalscope.rs
+++ b/components/script/dom/worklet/paintworkletglobalscope.rs
@@ -24,7 +24,7 @@ use net_traits::image_cache::ImageCache;
 use pixels::PixelFormat;
 use script_bindings::cell::DomRefCell;
 use script_bindings::interfaces::HasOrigin;
-use script_bindings::reflector::{DomObject, dom_object_value};
+use script_bindings::reflector::dom_object_value;
 use script_traits::{DrawAPaintImageResult, PaintWorkletError, Painter};
 use servo_base::id::PipelineId;
 use servo_config::pref;
@@ -334,9 +334,7 @@ impl PaintWorkletGlobalScope {
         rooted!(&in(cx) let argument_object = unsafe { NewArrayObject(cx, &arguments_value_array) });
 
         rooted_vec!(let mut callback_args);
-        callback_args.push(ObjectValue(
-            rendering_context.reflector().get_jsobject().get(),
-        ));
+        callback_args.push(dom_object_value(&*rendering_context));
         callback_args.push(dom_object_value(&*paint_size));
         callback_args.push(dom_object_value(&*properties));
         callback_args.push(ObjectValue(argument_object.get()));
